### PR TITLE
updated nirmata_test.sh

### PR DESCRIPTION
**Pull request by rootxrishabh**
This PR is for the following changes -

- Add support for MongoDB 5.0.15.
- Add check for Kafka controllers.
- Remove zookeeper check of Kafka brokers through `/brokers/ids` as path is unavailable.

### CODE REVIEW

File: nirmata_test/nirmata_test.sh [hunk 1]
```diff
@@ -666,58 +505,52 @@ done
...
-  fi
+kafka_controller_test(){
+echo "Testing Kafka controller pods"
+kafka_controller_ns=$kafka_ns
+kafkas_controllers=$(kubectl get pod -n $kafka_controller_ns -l nirmata.io/service.name=kafka-controller --no-headers | awk '{print $1}')
+kaf_cont_num=0
+kaf_cont_error=0
+for kafka_controller in $kafkas_controllers; do
+    echo "Found Kafka Controller Pod $kafka_controller"
+    kafka_controller_df=$(kubectl -n $kafka_controller_ns exec $kafka_controller -- df /var/lib/kafka | awk '{ print $5; }' |tail -1|sed s/%//)
```
Comment (atulatnirmata) [inline nirmata_test/nirmata_test.sh:523]: Be consistent with spacing.
Somewhere you have used `df /var/lib/kafka | awk` somewhere else you have used `awk '{ print $5; }' |tail -1|`
Putting one space before and after an operator helps readability.
In the above two examples, the first one is more readable than the second. Please incorporate this everywhere in your code.

File: nirmata_test/nirmata_test.sh [hunk 2]
```diff
@@ -666,58 +505,52 @@ done
...
+    [[ $kafka_controller_df -gt $df_free ]] && error "Found Kafka volume at ${kafka_df}% usage on $kafka_controller"
+    kaf_cont_num=$((kaf_cont_num + 1));
```
Comment (atulatnirmata) [inline nirmata_test/nirmata_test.sh:525]: I don't see this variable being set before. It is inited with 0. What is the use of this line?

File: nirmata_test/nirmata_test.sh [hunk 3]
```diff
@@ -666,58 +505,52 @@ done
...
-
-  if [[ $error_count -eq 0 ]]; then
-    good "All deployments passed tests"
-  else
-    error "Some deployments failed tests"
-  fi
+kafka_controller_test(){
+echo "Testing Kafka controller pods"
+kafka_controller_ns=$kafka_ns
+kafkas_controllers=$(kubectl get pod -n $kafka_controller_ns -l nirmata.io/service.name=kafka-controller --no-headers | awk '{print $1}')
```
Comment (atulatnirmata) [inline nirmata_test/nirmata_test.sh:518]: Change variable name from kafkas_controllers to kafka_controllers

File: nirmata_test/nirmata_test.sh [hunk 4]
```diff
@@ -666,58 +505,52 @@ done
...
         cp -f /certs/*.crt /usr/local/share/ca-certificates/
         update-ca-certificates
     fi
     [ -z $logfile ] && logfile="/tmp/k8_test.$$"
     [ -z $EMAIL_USER ] && EMAIL_USER="" #would this ever work?
     [ -z $EMAIL_PASSWD ] && EMAIL_PASSWD="" #would this ever work?
-    [ -z "$TO" ] && error "No TO address given!!!" && exit 1 # Why did I comment this out?
-    [ -z "$SUBJECT" ] && SUBJECT="K8 test script error" && echo -e "\e[33mYou provided no Subject using $SUBJECT \e[0m"
-    # This needs to be redone with less nesting and more sanity.
+    #[ -z $TO ] && error "No TO address given!!!" && exit 1 # Why did I comment this out?
```
Comment (atulatnirmata) [inline nirmata_test/nirmata_test.sh:549]: Remove commented code

File: nirmata_test/nirmata_test_old.sh
Comment (atulatnirmata) [inline nirmata_test/nirmata_test_old.sh:1]: Remove this file if it won't be used. No need of adding a new file to the repo.

File: nirmata_test/nirmata_test.sh [hunk 5]
```diff
@@ -1284,4 +1070,4 @@ if [ $warn != 0 ];then
 fi
 echo -e  "\e[32mTesting completed without errors or warning\e[0m"
 do_email
-exit 0
+exit 0
```
Comment (atulatnirmata) [inline nirmata_test/nirmata_test.sh:1073]: Add blank line at end of file. Read this to understand why a blank line should be added at end of file https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file

**Review by atulatnirmata — COMMENTED** [review]
I have left comments. I have only reviewed the kafka part and not the entire script. Some comments that I have left related to indentation and style are valid for the entire script.

**Review by vikashkaushik01 — APPROVED** [review]

File: nirmata_test/nirmata_test.sh (hunk 1 above)
Comment (rootxrishabh) [reply nirmata_test/nirmata_test.sh:523]: Spaces between operators are not allowed in bash scripts. 
We are not using `df /var/lib/kafka | awk` and `awk '{ print $5; }' |tail -1|` separately. They are being used as a combined statement everywhere.

Comment (atulatnirmata) [reply nirmata_test/nirmata_test.sh:523]: Just ensure consistent spacing between pipe i.e. one space before and after the pipe symbol
`kafka | awk | echo foo`

File: nirmata_test/nirmata_test.sh [hunk 6]
```diff
@@ -521,43 +436,12 @@ zoos=$(kubectl get pod -n $zoo_ns -l 'nirmata.io/service.name in (zookeeper, zk)
...
-        warn $zoo node count is $zoo_node_count
-        kubectl exec $zoo -n $zoo_ns --  sh -c "echo srvr | nc localhost 2181"
-    fi
-    # High latency indicate a resource issue
-    zoo_latency=$(kubectl exec $zoo -n $zoo_ns -- sh -c "echo srvr | nc localhost 2181" |grep Latency |tr "/" " " |awk '{ print $6; }')
-    if [ $zoo_latency -lt $zoo_latency_max ];then
-        good $zoo node latency is $zoo_latency
+    curr_zoo=$(kubectl -n $zoo_ns exec $zoo -- sh -c "/opt/apache-zookeeper-*/bin/zkServer.sh status" 2>&1|grep Mode)
+    zoo_node_count=$(kubectl exec $zoo -n $zoo_ns -- sh -c "echo srvr | nc localhost 2181|grep Node.count:" |awk '{ print $3; }')
+    if [[ $zoo_node_count -lt 50000 ]];then
```
[the remark this answers is not in the document]
Comment (rootxrishabh) [reply nirmata_test/nirmata_test.sh:441]: @atulatnirmata I had asked Anubhav about this. We noticed that whenever the znode count passes the 50000 mark, the zookeeper performance used to drop a lot. That's why we have kept it to 50000 as a check.

**Review by atulatnirmata — APPROVED** [review]
looks good

**Pull request merged**